pkg/cloud/azurestack/azurestack.go

```diff
@@ -86,3 +90,38 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
 	}
 	return assets, nil
 }
+
+// IsAzureStackHub inspects the infrastructure config returns true if it is configured for ASH.
+func IsAzureStackHub(platformStatus *configv1.PlatformStatus) bool {
+	return platformStatus.Azure != nil && platformStatus.Azure.CloudName == configv1.AzureStackCloud
+}
+
+// CloudConfigTransformer implements the cloudConfigTransformer. It takes
+// the user-provided, legacy cloud provider-compatible configuration and
+// modifies it to be compatible with the external cloud provider. It returns
+// an error if the platform is not OpenStackPlatformType or if any errors are
+// encountered while attempting to rework the configuration.
+func CloudConfigTransformer(source string, infra *configv1.Infrastructure, network *configv1.Network) (string, error) {
+	if !IsAzureStackHub(infra.Status.PlatformStatus) {
+		return "", fmt.Errorf("invalid platform, expected CloudName to be %s", configv1.AzureStackCloud)
+	}
+
+	var cfg azure.Config
+	if err := json.Unmarshal([]byte(source), &cfg); err != nil {
+		return "", fmt.Errorf("failed to unmarshal the cloud.conf: %w", err)
+	}
+
+	// If the virtual machine type is not set we need to make sure it uses
+	// the "standard" instance type. This is to mitigate an issue in the 1.27
+	// release where the default instance type was changed to VMSS.
+	// see OCPBUGS-20213 for more information.
+	if cfg.VMType == "" {
+		cfg.VMType = vmTypeStandard
+	}
+
+	if cfgbytes, err := json.Marshal(cfg); err != nil {
+		return "", fmt.Errorf("failed to marshal the cloud.conf: %w", err)
+	} else {
+		return string(cfgbytes), nil
+	}
```

Comment:
Nit, I'm sure there's a linter that would suggest removing this else 🤔 
```suggestion
	}
	
	return string(cfgbytes), nil
```

Reply:
but then i also need to take `cfgbytes` out of the `if` clause assignement right?

Reply:
Yeah you would need to, I think that's more idiomatic from my understanding